### PR TITLE
Piglatinfunc

### PR DESCRIPTION
**Pull request by Kaylotura**

### CODE REVIEW

File: practice/pig-latin.py [hunk 1]
```diff
@@ -1,58 +1,122 @@
 import string
```
Comment (davidselassie) [inline practice/pig-latin.py]: Module docstring goes before imports.

File: practice/pig-latin.py [hunk 2]
```diff
@@ -1,58 +1,122 @@
...
 """
 
-# 1. Setup
-FIRST_VOWEL = 'a e i o u'.split()
-LATER_VOWEL = 'a e i o u y'.split()
+
+# 1. Setup Define/Functions
+
+
+FIRST_VOWEL = 'a e i o u A E I O U'.split()
```
Comment (davidselassie) [inline practice/pig-latin.py]: We'll go over this more later, but I suggest having variable names parallel the structure within. So for variables that contain lists, have the variable name be plural as a hint that there are multiple items inside.

File: practice/pig-latin.py [hunk 3]
```diff
@@ -1,58 +1,122 @@
...
+LATER_VOWEL = 'a e i o u y A E I O U Y'.split()
+SPECIAL_CASE = 'Ybe ybe Ycleped ycleped Yclept yclept Ysgard Ydo ydo Ydrad ydrad Yfere yfere Ygdrasyl ygdrasyl Yghe'\
+               'ygh Ygo ygo Yground yground Ypres ypres Ypsiliform ypsiliform Ypsiloid ypsiloid Yren yren Ysame ysame '\
+               'Ytterbic ytterbic Ytterbium ytterbium Yttria yttria Yttric yttric Yttriferous yttriferous Yttrious ' \
+               'yttrious Yttrium yttrium Yttro-cerite yttro-cerite Yttro-columbite yttro-columbite Yttro-tantalite ' \
+               'yttro-tantalite Yvel yvel Ywar ywar Ywis ywis'.split()
 PUNCTUATION_LIST = list(string.punctuation)
 CAPITALS_LIST = list(string.ascii_uppercase)
```
Comment (davidselassie) [inline practice/pig-latin.py]: Similarly, I think adding `_list` is unnecessary if the name is plural.

You might be wanting to distinguish from a string of all the capital letters, but I don't think you're performing any operations where the distinction matters.

File: practice/pig-latin.py [hunk 4]
```diff
@@ -1,58 +1,122 @@
...
+    else:
+        return False
 
-# 3. Transform
+def check_for_and_cut_capitals(english_word):
+    """Checks for capitals at the start of a word, and casts it to lowercase, checking capitalize_boolean true"""
+    uppercase_first_letter = english_word[0]
+    lowercase_first_lettter = uppercase_first_letter.lower()
+    strip_capital_english_word = english_word.replace(uppercase_first_letter,lowercase_first_lettter,1)
+    return strip_capital_english_word
```
Comment (davidselassie) [inline practice/pig-latin.py]: Why not just return `english_word.lower()`? To handle words that have special internal capital letters?

File: practice/pig-latin.py [hunk 5]
```diff
@@ -1,58 +1,122 @@
...
+# 2. Define Functions
+
+
+def get_phrase_casted_into_list():
+    """Aquires english phrase from user input and casts it as a variable"""
+    typed_phrase = input("What would you like to Anslatetray? ").split()
+    return typed_phrase
+
+
+def pull_english_word_from_list (phrase, i):
```
Comment (davidselassie) [inline practice/pig-latin.py]: I think this falls over the line of being a function that is extraneous. The subscript operator is only used for picking a specific item out of a list, and so it wouldn't be mis-interpreted as anything else and I think giving it a name is extraneous. That's okay! Honing in on the balance between too many functions and too few functions is hard.

File: practice/pig-latin.py [hunk 6]
```diff
@@ -1,58 +1,122 @@
...
-    if english_word[-1] in PUNCTUATION_LIST:
-        new_punctuation = english_word[-1]
-        modified_english_word = english_word.strip(new_punctuation)
-        re_capitlaize_word = False
-    elif english_word[0] in CAPITALS_LIST:
-        modified_english_word = english_word.lower()
-        re_capitlaize_word = True
+def flip_punctuation_boolean(word):
+    """Checks for punctuation at end of word and flips a boolean switch if it is present """
+    if word[-1] in PUNCTUATION_LIST:
```
Comment (davidselassie) [inline practice/pig-latin.py]: When you have something that looks like:

```
if ...:
    return True
else:
    return False
```

You could more simply write `return ...` so in this case `return not word[-1] in PUNCTUATION_LIST`.

File: practice/pig-latin.py [hunk 7]
```diff
@@ -1,58 +1,122 @@
...
+    punctuation = english_word[-1]
+    return punctuation
+
+def strip_punctuation(english_word):
+    """cuts punctuation of english word"""
+    return english_word[0:-1]
+
+def convert_english_word_to_pig_latin_word(english_word):
+    """Translates english_word into pig_latin_word considering for up to 4 consonants in a row."""
+    if (english_word[0] in FIRST_VOWEL):
```
Comment (davidselassie) [inline practice/pig-latin.py]: Extraneous parens.

File: practice/pig-latin.py [hunk 8]
```diff
@@ -1,58 +1,122 @@
...
 
-# 4. Print
 
-print(new_phrase)
\ No newline at end of file
+def main():
+    """Main Function"""
+    pig_latin_phrase = ''
+    english_phrase = get_phrase_casted_into_list()
+    for english_word in english_phrase:
```
Comment (davidselassie) [inline practice/pig-latin.py]: Break apart this function further: always solve the item-level problem separately and before the list-level problem and is a specific step as part of the main. Work on the sub problem of converting a single word to Pig Latin as it's own problem first, then the list level, then call that in the main.

File: practice/pig-latin.py (hunk 4 above)
Comment (Kaylotura) [inline practice/pig-latin.py]: That was my reasoning behind using the longer function.

File: practice/pig-latin.py [hunk 9]
```diff
@@ -1,58 +1,180 @@
...
+    """
+    # for loop? List comprehension?
+
+def translate_to_pig_latin(words):
+    """Translates each word in words to pig latin
+    >>> translate_to_pig_latin(['i','like','to','eat','bananas,', 'mackleMore', 'mcDonalds.'])
+    ['iyay','ikelay','otay','eatyay','ananasbay,', 'ackleMoreay', 'onaldsmcDay']
+    """
+
+    for word in words, latin_func(word) = ???
```
Comment (davidselassie) [inline practice/pig-latin.py:57]: This code looks incomplete? Did we forget some commits in your branching fix up? Let me know.

File: practice/pig-latin.py [hunk 10]
```diff
@@ -1,58 +1,180 @@
...
+#         if punctuation_boolean == True:
+#             temp_pig_latin_word = paste_punctution_back_on_pig_latin_word (temp_pig_latin_word,saved_punctuation)
+#         pig_latin_phrase += temp_pig_latin_word + ' '
+#     print_full_pig_latin_phrase(pig_latin_phrase)
+# main()
+
+
+
+
+def main ():
```
Comment (davidselassie) [inline practice/pig-latin.py:171]: Looks like this was an in-progress version of the code. Find me and we can find if there are any commits that are not in this branch that contain the correct content.

**Pull request merged**